### PR TITLE
feat: beta test preparation

### PR DESCRIPTION
**Pull request by nullishamy**
This gets the project ready for its beta test. Specifically, this adds a create command and refactors the config to support configurable testing guilds.

### CODE REVIEW

**Comment by nullishamy** [comment]
Once #36 is merged, we can consider review, merge and deployment of this.

**Comment by nullishamy** [comment]
With #36 merged, we can proceed with review of this PR.

File: src/managers/commands.ts [hunk 1]
```diff
@@ -91,21 +91,33 @@ export class CommandsManager extends Manager {
...
 
     for (const command of commands) {
       const builder = new SlashCommandBuilder()
 
       builder.setName(command.name)
         .setDescription(command.description)
 
-      if (command.guildId) {
+      if (this.client.config.nodeEnv() === 'development' && !command.guildId) {
+        // We dont specify a guild, default to testing
```
Comment (BobobUnicorn) [inline src/managers/commands.ts]:
```suggestion
        // In development environments, default all commands to be registered to the testing guild; this
        // allows us to actually refresh more often than once per hour.
```

File: src/managers/commands.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export interface Command {
   onAutocomplete?: (client: ProjectsClient, interaction: AutocompleteInteraction<'cached'>) => Awaitable<Result<void, Error>>
   onModalSubmit?: (client: ProjectsClient, interaction: ModalSubmitInteraction<'cached'>) => Awaitable<Result<void, Error>>
 
-  guildId?: Snowflake
+  shouldPublish: boolean
```
Comment (BobobUnicorn) [inline src/managers/commands.ts]:
```suggestion
  shouldPublishGlobally: boolean;
```

File: src/managers/commands.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export interface Command {
   onAutocomplete?: (client: ProjectsClient, interaction: AutocompleteInteraction<'cached'>) => Awaitable<Result<void, Error>>
   onModalSubmit?: (client: ProjectsClient, interaction: ModalSubmitInteraction<'cached'>) => Awaitable<Result<void, Error>>
 
-  guildId?: Snowflake
+  shouldPublishGlobally: boolean
```
Comment (BobobUnicorn) [inline src/managers/commands.ts:24]:
```suggestion
  shouldPublishGlobally: boolean;
```

Comment (BobobUnicorn) [reply src/managers/commands.ts:24]: :)

**Review by BobobUnicorn — APPROVED** [review]

**Pull request merged**